### ISSUE TITLE
Issues with minDate and maxDate

### ISSUE THREAD

**Issue by malreynolds** [zippyui/react-date-picker#96]
Hi, I'm using a DateField and I'm explicitly specifying minDate and maxDate, forceValidDate is on

If I have the value "29/01/2016" in the DateField, If I start entering anything that's not a valid date (for example if the cursor is over the "2" of "2016" and I enter 1, then I start seeing the following error messages in the console

> main-c5ab414….js:126992 Uncaught ReferenceError: moment is not defined****

I had a look at the bundled code and the code that's causing this is

```
	      if (minDate && dateMoment.isBefore(minDate)) {
	        var clone = moment(dateMoment);
```

I had a look at the source and here's what I think is the faulty line:

https://github.com/zippyui/react-date-picker/blob/fda854ce6c8bcdf5c198e636b2049721a08aa506/src/DateFormatInput/index.js#L327

```
    if (minDate && dateMoment.isBefore(minDate)) {
      const clone = moment(dateMoment)
```

I suspect this needs to be a  `this.toMoment(dateMoment)` call, instead of `moment(dateMoment)`, as moment is not defined in the file.

My other issue is to do with with the Navbar - if my minDate is 20th of May, but the current date is 20th June, the calendar will show June, but the back arrow on the navbar wouldn't allow me to go back another month. Same story with maxDate.

Thanks!

**Comment by radubrehar** [comment]
Fixed in `3.1.12` which is on npm already! Thanks for reporting!

**Comment by malreynolds** [comment]
Great stuff, works like a charm now. 

Final question - is there a way to completely disable the input field (if not, can you add one). I feel like the current way of handling invalid input is a bit stiff(for example if you enter invalid input, the field just corrects you, according to its understanding of correct). I would rather have the users be forced to use the date picker at this point, until you can come up with some way (maybe a tooltip?) to notify the user what they are doing wrong.

**Comment by radubrehar** [comment]
I don't exactly understand what you need. But you can use a `<Calendar />` separately from a `<DateField />` if that helps.
You can also play with `forceValidDate` and `validateOnBlur` (which is useful when `forceValidDate` is false). You can also specify the desired `dateFormat` to the field, which supports all moment.js formats

**Comment by malreynolds** [comment]
I was talking about the ability to disable the input field.

Example here (last name is disabled): 
http://www.w3schools.com/tags/tryit.asp?filename=tryhtml_input_disabled.

**Comment by malreynolds** [comment]
https://github.com/zippyui/react-input-field - looks like react-input-field already supports disabled prop, so potentially can't you just propagate that from DateField to the inner Input?

**Comment by radubrehar** [comment]
I'll make sure the `disabled` prop works as expected.

**Comment by radubrehar** [comment]
Just released `5.3.15` on npm with the fix!

**Comment by malreynolds** [comment]
Now works as expected.
